### PR TITLE
[IMP] IAP/CRM : lead enrichment 3.0

### PR DESCRIPTION
**Pull request by fleodoo**
Before this commit it is not possible to enrich leads manually,
only a cron enriches leads.
After this commit and option is added in the settings to chose if
you want to either use the cron or to only enrich manually
If you chose to use the manual mode, a button appears on the lead
to enrich if certain conditions are met (email_adress present, not
already enriched,...)

task-id: 2075817

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/crm_iap_lead_enrich/models/iap_enrich_api.py [hunk 1]
```diff
@@ -14,9 +14,10 @@ class IapEnrichAPI(models.AbstractModel):
     def _contact_iap(self, local_endpoint, params):
         account = self.env['iap.account'].get('reveal')
         dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('reveal.endpoint', self._DEFAULT_ENDPOINT)
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/models/iap_enrich_api.py]: Pro or Rar asked me to remove it, is it back ?

File: addons/crm_iap_lead_enrich/views/crm_lead_enrich_views.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="crm_lead_enrich" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/views/crm_lead_enrich_views.xml]: Could you update according to guidelines ?
id: crm_lead_view_form
name: crm.lead.view.form.inherit.iap.lead.enrich
...
Filename: crm_lead_views.xml

File: addons/crm_iap_lead_enrich/views/crm_lead_enrich_views.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <field name="name">crm.lead.enrich</field>
+        <field name="model">crm.lead</field>
+        <field name="inherit_id" ref="crm.crm_lead_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='%(crm.action_crm_lead2opportunity_partner)d']" position="after">
+                <field name="show_enrich_button" invisible="1"/>
+                <field name="type" invisible="1"/>
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/views/crm_lead_enrich_views.xml]: Are you sure it is not already present in the view ? Seems weird.

File: addons/crm_iap_lead_enrich/models/crm_lead.py [hunk 4]
```diff
@@ -123,21 +122,11 @@ def _iap_enrich_get_message_data(self, company_data):
...
-        mail_template = self.env.ref('crm_iap_lead_enrich.mail_template_data_iap_lead_enrich_nocredit')
-        if not mail_template:
-            return
-
-        mail_template.sudo().send_mail(iap_account.id, force_send=False, notif_layout='mail.mail_notification_light')
-        self.env['ir.config_parameter'].sudo().set_param('lead_enrich.already_notified', True)
+    @api.depends('email_from', 'probability', 'iap_enrich_done', 'reveal_id')
+    def _compute_show_enrich_button(self):
+        for lead in self:
+            config = self.env['ir.config_parameter'].sudo().get_param('crm.lead_enrichment_auto','auto')
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/models/crm_lead.py]: Move the config parameter check outside of the loop. Even if it is cached it is a good pratice to separate static from dynamic computation.

File: addons/crm_iap_lead_enrich/models/crm_lead.py [hunk 5]
```diff
@@ -123,21 +122,11 @@ def _iap_enrich_get_message_data(self, company_data):
...
+            if lead.active == False or config != 'on_demand' or lead.email_from == False or lead.iap_enrich_done or lead.reveal_id != False or lead.probability in (0, 100):
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/models/crm_lead.py]: Actually no need to loop on leads if config is not on_demand -> if not config or config != 'on_demande': directly set to False

File: addons/crm_iap_lead_enrich/models/iap_enrich_api.py (hunk 1 above)
Comment (fleodoo) [reply addons/crm_iap_lead_enrich/models/iap_enrich_api.py]: removed it, I used it when testing myself.

File: addons/crm_iap_lead_enrich/models/crm_lead.py (hunk 5 above)
Comment (fleodoo) [reply addons/crm_iap_lead_enrich/models/crm_lead.py]: actually no need to loop as all as there is only one lead

Comment (tde-banana-odoo) [reply addons/crm_iap_lead_enrich/models/crm_lead.py]: Well compute methods should always work in batch, as it depends on how you modify (triggers will define the set of records to recompute) ... consider self as a true record set with potentially millions of records ^^ .

File: addons/crm/models/res_config_settings.py [hunk 6]
```diff
@@ -14,6 +14,10 @@ class ResConfigSettings(models.TransientModel):
     module_crm_iap_lead = fields.Boolean("Generate new leads based on their country, industries, size, etc.")
     module_crm_iap_lead_website = fields.Boolean("Create Leads/Opportunities from your website's traffic")
     module_crm_iap_lead_enrich = fields.Boolean("Enrich your leads automatically with company data based on their email address.")
+    lead_enrich_auto = fields.Selection([
```
[the remark this answers is not in the document]
Comment (fleodoo) [reply addons/crm/models/res_config_settings.py]: I'm not sure my correction is what you asked, could you please verify?

**Comment by tde-banana-odoo** [comment]
@robodoo r+ p=1

**Pull request closed without merging**